### PR TITLE
Fix to char-

### PR DESCRIPTION
**Pull request by akeep**
Small fix to #627 so that `char-` will return a negative number instead of a large positive integer when the second argument is `char>?` the first argument.

Also added a couple of tests for `char-` to the mats.

### CODE REVIEW

File: LOG [hunk 1]
```diff
@@ -2309,3 +2309,8 @@
 - added install-docs target for make and corresponding configuration
   options: --installdoc, --installcsug, and --installreleasenotes
     configure makefiles/Makefile.in
+- fixed a bug in `char-` that lead to returning a large positive integer
```
Comment (burgerrg) [inline LOG:2312]: `lead` => `led`

**Review by burgerrg — COMMENTED** [review]
Thanks for this fix! Please update the release notes as well.

**Comment by akeep** [comment]
Thanks for having a look @burgerrg!  I've updated the release notes and fixed the log.  If it all looks good I'll squash and merge.

**Review by burgerrg — APPROVED** [review]
Thanks for fixing this bug!

**Pull request merged**